dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java

```diff
@@ -80,6 +81,7 @@ protected final void service(HttpServletRequest req, HttpServletResponse respons
         final VelocityRequestWrapper request =VelocityRequestWrapper.wrapVelocityRequest(req);
         final String uri = CMSUrlUtil.getCurrentURI(request);
         final boolean comeFromSomeWhere = request.getHeader("referer") != null;
+        HttpServletRequestThreadLocal.INSTANCE.setRequest(request);
```

Comment:
why do we need to set it here if we are setting it into the CMSFIlter?

https://github.com/dotCMS/core/blob/9a35a7cb4ad30a4be34cca88e66415d9a100ac6a/dotCMS/src/main/java/com/dotmarketing/filters/CMSFilter.java#L65

Reply:
Also if we now are going to set the request here we still need to set it into the VelocityLiveMode

https://github.com/dotCMS/core/blob/1a30bc0efe612d6d55101330a5840659fb005c1d/dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java#L191